src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx

```diff
@@ -60,17 +63,29 @@ export function OpenSavedQueryFlyout({
   const [selectedQuery, setSelectedQuery] = useState<SavedQuery | undefined>(undefined);
   const [searchQuery, setSearchQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
 
+  const currentTabIdRef = useRef(selectedTabId);
```

Comment:
Can we add unit tests for this file?

Reply:
UT's for saved queries are all in this PR: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8798

Reply:
Added unit tests for this file in this commit: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8848/commits/c73d3579b3c97c6cef0cc77a670865a1fabe06eb